tests/orm/node/test_node.py

```diff
@@ -54,6 +54,14 @@ def setUp(self):
         super().setUp()
         self.node = Data()
 
+    def test_equality(self):
+        """Test the `__eq__` method."""
+        node1 = Node()
+        node2 = node1
```

Comment:
This test doesn't work quite as intended: Because `node1` and `node2` are the same Python object (`id(node1) == id(node2)`), `node1 == node2` will be `True` even in the current code.

Instead, I think we need to store a `Data` node and load it back, such that `node1` and `node2` are different Python objects.

That can be tested by ``assert node1 is not node2``.

Reply:
See the following (current behavior, before the applied fix):

```python
In [1]: from aiida.orm import Data

In [2]: d1 = Data().store()

In [3]: d2 = d1

In [4]: d3 = load_node(d1.uuid)

In [5]: d1 == d2
Out[5]: True

In [6]: d1 is d2
Out[6]: True

In [7]: d1 == d3
Out[7]: False

In [8]: d1 is d3
Out[8]: False
```

Reply:
Thanks! I had a slight suspicion of this but I found the problems with the intrinsic python-mock node types before I could get to that. I would anyways have had to ask you guys how to check this for sure, haha.